libbeat/metric/system/process/process.go

```diff
@@ -409,6 +388,53 @@ func (procStats *Stats) Get() ([]common.MapStr, error) {
 	return procs, nil
 }
 
+// GetOne fetches process data for a given PID if its name matches the regexes provided from the host.
+func (procStats *Stats) GetOne(pid int) (common.MapStr, error) {
+	if len(procStats.Procs) == 0 {
+		return nil, nil
+	}
+
+	newProcs := make(ProcsMap, 1)
+	p, err := procStats.getSingleProcess(pid, newProcs)
+	if err != nil {
+		return nil, err
+	}
+
+	e := procStats.getProcessEvent(p)
+
+	return e, nil
+}
+
+func (procStats *Stats) getSingleProcess(pid int, newProcs ProcsMap) (*Process, error) {
+	var cmdline string
+	var env common.MapStr
+	if previousProc := procStats.ProcsMap[pid]; previousProc != nil {
+		if procStats.CacheCmdLine {
+			cmdline = previousProc.CmdLine
+		}
+		env = previousProc.Env
+	}
+
+	process, err := newProcess(pid, cmdline, env)
+	if err != nil {
+		return nil, fmt.Errorf("Skip process pid=%d: %v", pid, err)
```

Comment:
Error string should not be capitalized, https://github.com/golang/go/wiki/CodeReviewComments#error-strings